### PR TITLE
feat(web): zone specific datetimes

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
AVO-586
It's not always valuable to have a zone date time formatted for a users location

## Description
This PR updates the date formatting to convert formatted dates to be zone specific, if no zone is selected it should revert to the users location. 

There were a few missing timezones in configs so I've also added those.

### CODE REVIEW

File: .github/workflows/CI.yml
Comment (VIKTORVAV99) [inline .github/workflows/CI.yml:542]: Is this really needed? Don't we specify the python version above?

Comment (tonypls) [reply .github/workflows/CI.yml]: No but why does a python test workflow fail 😢

Comment (VIKTORVAV99) [reply .github/workflows/CI.yml]: Hmm I can try removing the cache and see if that helps.

**Comment by cadeban** [comment]
Do we provide any sort of disclaimer about the data delay or will users come to the app and expect to see current time?
Screenshot 2024-10-14 at 14 28 11

**Comment by tonypls** [comment]
> Do we provide any sort of disclaimer about the data delay or will users come to the app and expect to see current time? Screenshot 2024-10-14 at 14 28 11

I think because the delay is unchanged we don't need to make a disclaimer right now. Hopefully we reduce the delay soon 🚀

**Comment by cadeban** [comment]
And what was the verdict on `GMT-X`?

**Review by cadeban — APPROVED** [review]
LGTM!

**Comment by cadeban** [comment]
Whoops, just seeing [Viktor's feedback](https://electricitymaps.slack.com/archives/C031YV8M940/p1728912933565289) in the channel now - maybe wait for his approval too

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

Seems like my syncing issues are unrelated to this.

**Pull request merged**
